### ISSUE TITLE
Allow rate limiting by input amount

### ISSUE THREAD

**Issue by mxstbr** [teamplanes/graphql-rate-limit#24]
* **I'm submitting a ...**
[ ] bug report
[x] feature request
[ ] question about the decisions made in the repository
[ ] question about how to use this project

* **Summary**

I want to rate limit an `inviteUsers(emails: [String!]!)` mutation by the amount of emails invited.

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, personal fork, etc.)

I would like to be able to specify the maximum rate limit by the length of an input array. Each call would be multiplied by the amount of input elements.

For example, if I want a user to only be able to invite 5000 emails per week, something like this:

```GraphQL
type Mutation {
  inviteUsers(emails: [String!]!): [Boolean!]! @rateLimit(field: "emails", max: 5000, window: "7d")
}
```

I do not know if `field` would be the right name for the argument, but I think you can understand what I mean. Does that make sense?

**Comment by mxstbr** [comment]
I think this could be implemented relatively quickly by storing `n` timestamps in `validateResolve`.

**Comment by kirkness** [comment]
Released in 1.2.4!
